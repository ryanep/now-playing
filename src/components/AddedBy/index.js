import React from "react";
import styles from "./style.scss";

export default ({ name, photo }) => (
  <div className={styles.info}>
    <div className={styles.added}>
      <div className={styles.addedName}>Added by</div>
      <div id="added-by-name" className={styles.name}>{name}</div>
    </div>
    {photo && <img className={styles.photo} src={photo} alt={name} />}
  </div>
);
