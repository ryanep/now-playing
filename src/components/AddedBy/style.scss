.info {
	min-width: 20rem;
	text-align: right;
}

.added {
    display: inline-block;
    vertical-align: middle;
    text-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
}

.addedName {
    font-size: 1.4rem;
}

.name {
    font-weight: bold;
}

.photo {
    display: inline-block;
    vertical-align: middle;
    margin-left: 1rem;
    width: 3.5rem;
    height: auto;
    border-radius: 0.4rem;
    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
}