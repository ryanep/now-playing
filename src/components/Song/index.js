import React from "react";
import styles from "./style.scss";

export default ({ title, artist, artwork, added }) => (
  <div className={styles.song}>
    <img src={artwork} className={styles.artwork} alt={title} />
    <div className={styles.track}>
      <h1 className={styles.title}>{title}</h1>
      <h2 className={styles.artist}>{artist}</h2>
    </div>
    {added &&
      <div className={styles.info}>
        <img
          className={styles.photo}
          src={added && added.photo}
          alt={added.name}
        />
        <div className={styles.added}>
          <div className={styles.addedName}>Added by</div>
          <div className={styles.name}>{added && added.name}</div>
        </div>
      </div>}
  </div>
);
