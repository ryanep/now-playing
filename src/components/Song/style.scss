.song {
    position: absolute;
    top: 50%;
    width: 100%;
    transform: translateY(-50%);
    text-align: center;
}

.album {
    position: relative;
    display: inline-block;
    margin-bottom: 3rem;

    &::after {
        position: absolute;
        left: 0;
        bottom: 0;
        width: 100%;
        height: 30rem;
        content: '';
        background: linear-gradient(to bottom, rgba(0,0,0,0) 0%,rgba(0,0,0,0.9) 100%);
    }
}

.artwork {
    display: block;
    width: 70vw;
    height: auto;
    max-width: 64rem;
    border-radius: 3px;
    box-shadow: 0 2px 20px rgba(0,0,0,0.5);
}

.track {
    position: absolute;
    bottom: 6rem;
    width: 100%;
    padding: 0 2rem;
    text-align: left;
    z-index: 2;
}

.title {
    margin: 0;
    font-size: 4.2rem;
    margin-bottom: 0.5rem;
    font-weight: 200;
    text-shadow: 0 2px 2px rgba(0,0,0,0.2);

    @media (max-width: 750px) {
        font-size: 3.6rem;
    }
}

.artist {
    margin: 0;
    font-size: 3.2rem;
    font-weight: 400;
    text-shadow: 0 2px 2px rgba(0,0,0,0.2);
}

.photo {
    display: inline-block;
    vertical-align: middle;
    margin-right: 2rem;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    box-shadow: 0 2px 2px rgba(0, 0, 0, 0.4);
}

.info {
    margin-bottom: 2rem;
}

.added {
    display: inline-block;
    vertical-align: middle;
    text-align: left;
    text-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
}

.addedName {
    font-size: 1.4rem;
    margin-bottom: 0.5rem;
}

.name {
    font-weight: bold;
}