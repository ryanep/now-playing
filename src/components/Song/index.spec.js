import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Song from './index';
import AddedBy from '../AddedBy/index';

describe('components/Song', () => {
  const track = {
    title: '__title__',
    artist: '__artist__',
    artwork: [{ url: '__url__' }, { url: '__placeholder__' }],
    added: { name: '__added_by_name__' },
    isPlaying: '__is_playing__',
    progress: '__progress__',
    duration: '__duration__'
  };
  let component;

  beforeEach(() => {
    component = shallow(<Song {...track} />);
  });

  it('renders correctly', () => {
    const component = renderer.create(<Song {...track} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes the src and placeholder props to ProgressiveImage', () => {
    const progressiveImage = component.find('ProgressiveImage');
    expect(progressiveImage.length).toEqual(1);
    expect(progressiveImage.prop('src')).toEqual('__url__');
    expect(progressiveImage.prop('placeholder')).toEqual('__placeholder__');
  });

  it('renders the track title', () => {
    expect(component.find('h1').text()).toEqual('__title__');
  });

  it('renders the track artist', () => {
    expect(component.find('h2').text()).toEqual('__artist__');
  });

  it('passes addedBy object to AddedBy component if present', () => {
    const addedBy = component.find(AddedBy);
    expect(addedBy.length).toEqual(1);
    expect(addedBy.prop('name')).toEqual('__added_by_name__');
  });

  it('does not render the AddedBy component if no addedBy object present', () => {
    component = shallow(<Song {...track} added={null} />);
    expect(component.find(AddedBy).length).toEqual(0);
  });

  it('passes isPlaying, progress and duration to Progress component', () => {
    const progress = component.find('Progress');
    expect(progress.length).toEqual(1);
    expect(progress.prop('isPlaying')).toEqual('__is_playing__');
    expect(progress.prop('progress')).toEqual('__progress__');
    expect(progress.prop('duration')).toEqual('__duration__');
  });

  it('passes isPlaying to Visualiser component', () => {
    const visualiser = component.find('Visualiser');
    expect(visualiser.length).toEqual(1);
    expect(visualiser.prop('isPlaying')).toEqual('__is_playing__');
  });
});
