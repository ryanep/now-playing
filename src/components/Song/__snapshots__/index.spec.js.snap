// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Song renders correctly 1`] = `
<div
  className="song"
>
  <div
    className="album"
  >
    <img
      alt="__artist__ - __title__"
      className="artwork"
      src="http://localhost/__url__"
    />
    <div
      className="track"
    >
      <h1
        className="title"
      >
        __title__
      </h1>
      <div
        className="trackInfo"
      >
        <h2
          className="artist"
        >
          __artist__
        </h2>
        <div
          className="info"
        >
          <div
            className="added"
          >
            <div
              className="addedName"
            >
              Added by
            </div>
            <div
              className="name"
              id="added-by-name"
            >
              __added_by_name__
            </div>
          </div>
        </div>
      </div>
    </div>
    <progress
      className="progress"
      max={NaN}
      value={NaN}
    />
    <div
      className="visualiser"
    >
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
      <div
        className="bar playing"
      />
    </div>
  </div>
</div>
`;
